pkg/types/types.go

```diff
@@ -75,54 +75,115 @@ const (
+	case string(RuntimeNamePodman):
+		return RuntimeNamePodman
+	}
+	return RuntimeNameUnknown
+}
+
+type BasicRuntimeMetadata struct {
+	// RuntimeName is the name of the container runtime. It is useful to distinguish
+	// who is the "owner" of each container in a list of containers collected
+	// from multiple runtimes.
+	RuntimeName RuntimeName `json:"runtimeName,omitempty" column:"runtimeName,minWidth:5,maxWidth:12"`
+
+	// ContainerID is the container ContainerID without the container runtime prefix. For
+	// instance, without the "cri-o://" for CRI-O.
+	ContainerID string `json:"containerId,omitempty" column:"containerId,width:13,maxWidth:64"`
+
+	// ContainerName is the container name. In the case the container runtime
+	// response with multiple containers, ContainerName contains only the first element.
+	ContainerName string `json:"containerName,omitempty" column:"containerName,template:container"`
+}
+
+func (b *BasicRuntimeMetadata) IsEnriched() bool {
+	return b.RuntimeName != RuntimeNameUnknown && b.RuntimeName != "" && b.ContainerID != "" && b.ContainerName != ""
+}
+
+type BasicK8sMetadata struct {
+	Namespace     string `json:"namespace,omitempty" column:"namespace,template:namespace"`
+	PodName       string `json:"podName,omitempty" column:"pod,template:pod"`
+	ContainerName string `json:"containerName,omitempty" column:"container,template:container"`
+}
+
+func (b *BasicK8sMetadata) IsEnriched() bool {
+	return b.Namespace != "" && b.PodName != "" && b.ContainerName != ""
+}
+
+type K8sMetadata struct {
+	Node string `json:"node,omitempty" column:"node,template:node"`
+
+	BasicK8sMetadata `json:",inline"`
 
 	// HostNetwork is true if the container uses the host network namespace
 	HostNetwork bool `json:"hostNetwork,omitempty" column:"hostnetwork,hide"`
 }
 
+type CommonData struct {
+	// Runtime contains the container runtime metadata of the container
+	// that generated the event
+	Runtime BasicRuntimeMetadata `json:"runtime,omitempty" column:"runtime" columnTags:"runtime"`
+
+	// K8s contains the Kubernetes metadata of the object that generated the
+	// event
+	K8s K8sMetadata `json:"k8s,omitempty" columnTags:"kubernetes"`
+}
+
 func (c *CommonData) SetNode(node string) {
-	c.Node = node
+	c.K8s.Node = node
 }
 
-func (c *CommonData) SetContainerInfo(pod, namespace, container string) {
-	c.Pod = pod
-	c.Namespace = namespace
+func (c *CommonData) SetContainerMetadata(k8s *BasicK8sMetadata, runtime *BasicRuntimeMetadata, setContainerInfo bool) {
+	c.K8s.PodName = k8s.PodName
+	c.K8s.Namespace = k8s.Namespace
+
+	// In some cases, we don't have enough information to determine the exact
+	// container where the event happened.
+	if setContainerInfo {
+		c.K8s.ContainerName = k8s.ContainerName
 
-	// Container may have been enriched before by other means, so don't delete it here,
-	// if the incoming info is empty
-	if container != "" {
-		c.Container = container
+		c.Runtime.RuntimeName = runtime.RuntimeName
```

Comment:
I see that this is called from `EnrichEventByNetNs` for Kubernetes containers. The container runtime for all containers should be the same, so I think it's safe to set it even if we don't know the exact container within a pod.